converter/internal/prometheusconvert/testdata/azure.river

```diff
@@ -53,14 +53,8 @@ prometheus.remote_write "default" {
 		url            = "http://remote-write-url1"
 		send_exemplars = false
 
-		queue_config {
-			capacity             = 2500
-			max_shards           = 200
-			max_samples_per_send = 500
-		}
+		queue_config { }
 
-		metadata_config {
-			max_samples_per_send = 500
-		}
```

Comment:
@erikbaranowski after the prometheus version upgrade, I believe the defaults set by prometheus library match those that we set in Flow Mode, thus they are no longer here.

I ran a debugger to be sure and took a look at what are the values right after we run this line:
```
promConfig, err := prom_config.Load(string(in), false, log.NewNopLogger())
```
And I can see that they match our [Flow Mode defaults](https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.remote_write/#queue_config-block):
![image](https://github.com/grafana/agent/assets/17101802/ea4cbd07-d827-4aba-8c5d-030c32b49819)

So I think this is an expected change.

It bring an interesting "issue": it matters what version of prometheus config are you migrating from... We should probably raise in documentation clearly that we support only the prometheus config version that matches our `go.mod`.

Reply:
it's probably worth a look why queue_config shows up at all now. We should also update the doc page which does specify a version currently.

https://grafana.com/docs/agent/next/flow/reference/cli/convert/

Reply:
1) I've fixed the docs now to point to right version of prom
2) The reason the empty blocks are printed out, I believe, is that in the default value of `endpoint`, the `queue_config` is `nil`, while when constructing the `endpoint` in conversion code, we instead set it to a struct with default values. See this debug screenshot:

![image](https://github.com/grafana/agent/assets/17101802/d367270d-dac8-4f7b-8963-ff6efb8c6d06)

I'm thinking that to solve it, we may need to call `reflect.New` and `SetToDefault()` on every field that is `nil` in the loop over fields above, but there may be better ways. The issue is that we detect it's a default only on a recursive call to `encodeFields`, after a block for `queue_config` is already added. We should detect it before we call into the recursive call, I think.

Reply:
I did some more testing and found a few potential issues and generally it's a bit hard to determine how things should work in such cases where we have nested blocks with pointers and defaults. See: https://github.com/grafana/agent/pull/4430